### PR TITLE
Return correct status for register tx

### PR DESCRIPTION
**Pull request by Doy-lee**
This fixes `Staking 0.000000000 for a total fee of 0.000000000. Is this okay?` to return the correct error message instead of thinking that the tx construction succeeded.

### CODE REVIEW

File: src/wallet/wallet2.cpp [hunk 1]
```diff
@@ -7306,6 +7307,7 @@ wallet2::stake_result wallet2::create_stake_tx(const crypto::public_key& service
     return result;
   }
 
+  assert(result.status != stake_result_status::invalid);
```
Comment (msgmaxim) [inline src/wallet/wallet2.cpp]: Are we relying on result to be default initialized to stake_result_status::invalid? How about we make it explicit?

File: src/wallet/wallet2.cpp [hunk 2]
```diff
@@ -7553,10 +7555,10 @@ wallet2::register_service_node_result wallet2::create_register_service_node_tx(c
       result.msg += e.what();
       return result;
     }
-
-    result.status = register_service_node_result_status::success;
-    return result;
   }
+
+  assert(result.status != register_service_node_result_status::invalid);
```
Comment (msgmaxim) [inline src/wallet/wallet2.cpp]: Same as above

File: src/wallet/wallet2.cpp (hunk 1 above)
Comment (Doy-lee) [reply src/wallet/wallet2.cpp]: Sure. Fixed

File: src/wallet/wallet2.cpp (hunk 2 above)
Comment (Doy-lee) [reply src/wallet/wallet2.cpp]: Fixed

**Pull request merged**